app/src/main/java/com/yoshitoke/weatheringwithyou/mvp/view/MapsActivity.kt

```diff
@@ -0,0 +1,95 @@
+package com.yoshitoke.weatheringwithyou.mvp.view
+
+import android.content.Intent
+import android.location.Address
+import android.location.Geocoder
+import android.nfc.NfcAdapter
+import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.Marker
+import com.google.android.gms.maps.model.MarkerOptions
+import com.google.gson.Gson
+import com.yoshitoke.weatheringwithyou.R
+import com.yoshitoke.weatheringwithyou.mvp.model.DataClass.City
+import kotlinx.android.synthetic.main.activity_maps.*
+
+
+class MapsMarkerActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnMapLongClickListener {
+
+    companion object {
+        var marker: Marker? = null
+        var mGoogleMap: GoogleMap? = null
+        var localCity: City? = null
```

Comment:
why we need to use static variables?

Reply:
you means i should use lateinit variables? Because GoogleMap.addMarker(ARGs) will return nullable type

Reply:
Pls move out of `companion object` brackets

Reply:
ok sir